### ISSUE TITLE
fileserver: Breadcrumbs are broken if directory name contains percent sign %

### ISSUE THREAD

**Issue by abdusco** [caddyserver/caddy#5143]
Breadcrumbs are generated without text if directory name contains a percent sign.
The bug is present in both

```
> caddy version
v2.6.1 h1:EDqo59TyYWhXQnfde93Mmv4FJfYe00dO60zMiEt+pzo=
```

and also the latest master 6bad878a22e048762262d6fabe2144cefaf4ca81

To replicate:  
Take this file tree 

```
> tree
.
└── dir with % in the name
    └── another
        └── with %
            └── sub
```

and run caddy with `caddy file-server --browse`

it creates this HTML:

```html
<header>
	<h1>
		<a href="../../../../">/</a><a href="../../../"></a>/<a href="../../">another</a>/<a href="../"></a>/<a href="">sub</a>/
	</h1>
</header>
```

Note the missing link text for directories with `%` signs in their name.

A test case to replicate the issue:

```golang
// put this in
// modules/caddyhttp/fileserver/browsetplcontext_test.go:21

func TestBreadcrumbs(t *testing.T) {
	testdata := []struct {
		path     string
		expected []crumb
	}{
		{"folder/with % sign/sub", []crumb{
			{Link: "../../", Text: "folder"},
			{Link: "../", Text: "with % sign"}, // .Text == "" here
			{Link: "", Text: "sub"},
		}},
```

**Comment by abdusco** [comment]
The problem seems to be caused by the use of `url.PathUnescape`. 

> It returns an error if any % is not followed by two hexadecimal digits

https://pkg.go.dev/net/url#PathUnescape

and it returns a non-nil err, but it is ignored.

https://github.com/caddyserver/caddy/blob/33f60da9f2d6edc5de550275f043c4262d23f6ca/modules/caddyhttp/fileserver/browsetplcontext.go#L160

**Comment by abdusco** [comment]
I don't have a solution in mind at the moment, but I can submit a fix for this issue when I have the time.

**Comment by abdusco** [comment]
()I need to work on my force-push addiction.)

**Comment by abdusco** [comment]
The same problem also exists here:

https://github.com/caddyserver/caddy/blob/6bad878a22e048762262d6fabe2144cefaf4ca81/modules/caddyhttp/fileserver/browsetplcontext.go#L95

and it causes title to be rendered as `<title>.</title>`

**Comment by mholt** [comment]
@abdusco Thanks for the report, sorry it took so long for me to fix this. Can you please verify that the fix in #5534 works for you? It worked well for me on Linux. :+1:
